mava/wrappers/episode_metrics.py

```diff
@@ -72,13 +79,31 @@ def step(
         timestep.extras["episode_metrics"] = {
             "episode_return": episode_return_info,
             "episode_length": episode_length_info,
+            "final_step": done,
         }
 
         state = RecordEpisodeMetricsState(
             env_state=env_state,
+            key=state.key,
             running_count_episode_return=new_episode_return * not_done,
             running_count_episode_length=new_episode_length * not_done,
             episode_return=episode_return_info,
             episode_length=episode_length_info,
         )
         return state, timestep
+
+def get_final_step_metrics(metrics: Dict[str, chex.Array]) -> Dict[str, chex.Array]:
+    """Get the metrics for the final step of an episode.
+
+    Note: this is not a jittable method.
+    """
+    is_final_ep = metrics.pop("final_step")
```

Comment:
Perhaps rename "final_step" to "is_terminal_step" or something similar for clarity?